server/test/role.spec.js

```diff
@@ -0,0 +1,225 @@
+        should(res.status).equal(200);
+        should(res.body.title).equal(testData.updateRole1.title);
+        done();
+      });
+    });
+
+    it(`should NOT allow an Admin user with VALID token UPDATE a 
+    NON-EXISTENT Role`, (done) => {
+      server.put(`/roles/${newRoleId + 300}`)
+      .set({ 'x-access-token': adminToken })
+      .send(testData.updateRole1)
+      .end((error, res) => {
+        should(res.status).equal(404);
+        should(res.body.success).equal(false);
+        done();
+      });
+    });
+
+    it(`should NOT allow any User (Admin, Regular...) with an INVALID token 
+    UPDATE a Role`,
+    (done) => {
+      server.put(`/roles/${newRoleId}`)
+      .set({ 'x-access-token': 'invalid token' })
+      .send(testData.updateRole1)
+      .end((error, res) => {
+        should(res.status).equal(403);
+        done();
+      });
+    });
+
+    it(`should NOT allow a NON-Admin User with a VALID token 
+    UPDATE a Role`,
+    (done) => {
+      server.put(`/roles/${newRoleId}`)
+      .set({ 'x-access-token': regularUserToken })
+      .send(testData.updateRole1)
+      .end((error, res) => {
+        should(res.status).equal(403);
+        done();
+      });
+    });
+  });
+
+  describe('Get', () => {
+    it('should allow an Admin User with VALID token get all Roles',
+    (done) => {
+      server.get('/roles')
+      .set({ 'x-access-token': adminToken })
+      .end((error, res) => {
+        should(res.status).equal(200);
+        res.body.should.be.a.Array();
+        done();
+      });
+    });
+
+    it('should validate that atleast "regular" and "admin" roles exist',
+    (done) => {
+      server.get('/roles')
+      .set({ 'x-access-token': adminToken })
+      .end((error, res) => {
+        should(res.status).equal(200);
+        should(res.body[0].id).equal(1);
+        should(res.body[0].title).equal('admin');
+        should(res.body[1].id).equal(2);
+        should(res.body[1].title).equal('regular');
+        done();
+      });
+    });
+
+    it(`should NOT allow any User (Admin, Regular, ...) with an INVALID 
+    token to get all Roles`, (done) => {
+      server.get('/roles')
+      .set({ 'x-access-token': 'invalid token' })
+      .end((error, res) => {
+        should(res.status).equal(403);
+        done();
+      });
+    });
+
+    it('should NOT allow a Non-Admin User with VALID token get all Roles',
+    (done) => {
+      server.get('/roles')
+      .set({ 'x-access-token': regularUserToken })
+      .end((error, res) => {
+        should(res.status).equal(403);
+        done();
+      });
+    });
+
+    it('should allow an Admin User with VALID token get a Role',
+    (done) => {
+      server.get(`/roles/${newRoleId}`)
+      .set({ 'x-access-token': adminToken })
+      .end((error, res) => {
+        should(res.status).equal(200);
+        done();
+      });
+    });
+
+    it('should allow an Admin User with VALID token get a NON-EXISTENT Role',
```

Comment:
I don't think one can get a `NON-EXISTENT` role and the test is checking for a `not found`. Hence, you may want to update the description to reflect this.